com.unity.render-pipelines.universal/Runtime/ForwardRendererData.cs

```diff
@@ -39,20 +40,45 @@ public sealed class ShaderResources
 
             [Reload("Shaders/Utils/MaterialError.shader")]
             public Shader materialErrorPS;
+
+            // Core blitter shaders, adapted from HDRP
+            // TODO: move to core and share with HDRP
+            [Reload("Shaders/Utils/CoreBlit.shader"), SerializeField]
+            internal Shader coreBlitPS;
+            [Reload("Shaders/Utils/CoreBlitColorAndDepth.shader"), SerializeField]
+            internal Shader coreBlitColorAndDepthPS;
+
+
+            [Reload("Shaders/CameraMotionVectors.shader")]
+            public Shader cameraMotionVector;
+
+            [Reload("Shaders/ObjectMotionVectors.shader")]
+            public Shader objectMotionVector;
         }
 
-        public ShaderResources shaders = null;
+        public ShaderResources shaders;
 
-        public PostProcessData postProcessData = null;
+        public PostProcessData postProcessData;
 
 #if ENABLE_VR && ENABLE_XR_MODULE
         [Reload("Runtime/Data/XRSystemData.asset")]
-        public XRSystemData xrSystemData = null;
+        public XRSystemData xrSystemData;
 #endif
 
+        [SerializeField] LayerMask m_OpaqueLayerMask;
+        [SerializeField] LayerMask m_TransparentLayerMask;
+        [SerializeField] StencilStateData m_DefaultStencilState; // This default state is compatible with deferred renderer.
+        [SerializeField] bool m_ShadowTransparentReceive;
+        [SerializeField] RenderingMode m_RenderingMode;
+        [SerializeField] DepthPrimingMode m_DepthPrimingMode; // Default disabled because there are some outstanding issues with Text Mesh rendering.
+        [SerializeField] bool m_AccurateGbufferNormals;
+        [SerializeField] bool m_ClusteredRendering;
+        [SerializeField] TileSize m_TileSize;
```

Comment:
Why are you adding these properties?

Reply:
Similar to the reason above, these have to be there to be copied over to the new serialized object, this was partially why before the conversion process didn’t retain any data set on the ForwardRenderer, now this list also needs to contain properties that are new to the UniversalRendererData.
This file should only be needed for a brief time, since it is only needed for those taking Alpha/Beta 21.2 projects forward, I imagine we will be able to remove this upgrader in Unity 2022 lifecycle.